client/state/document-head/reducer.js

```diff
@@ -18,7 +18,7 @@ import {
 export function title( state = '', action ) {
 	switch ( action.type ) {
 		case DOCUMENT_HEAD_TITLE_SET:
-			return action.title;
+			return action.title || state;
```

Comment:
What about a separate action creator and action type specifically for unsetting the title?